packages/editor/src/state/callbacks.ts

```diff
@@ -222,6 +222,27 @@ export const useDownloadTrio = () =>
     }
   });
 
+export const useCopyToClipboard = () =>
+  useRecoilCallback(({ set, snapshot }) => async () => {
+    const workspace = snapshot.getLoadable(currentWorkspaceState)
+      .contents as Workspace;
+    const sub = workspace.files.substance.contents;
+    const sty = workspace.files.style.contents;
+    const dsl = workspace.files.domain.contents;
+    const concatenated = `-- .substance \n ${sub} \n 
```

Comment:
This formatting actually introduces extra spaces (see below). I think it'll be okay if you remove the spaces around all new lines
![image](https://github.com/penrose/penrose/assets/11740102/132f0963-a903-4882-847b-aa7267f90a10)

Reply:
Thanks for the catch! Fixed & tested in latest commit.